src/pal/inc/unixasmmacrosarm.inc

```diff
@@ -169,6 +173,7 @@ C_FUNC(\Name\()_End):
 
         free_stack __PWTB_StackAlloc
         POP_CALLEE_SAVED_REGISTERS
+	// TODO: From this point, r7 is not consistent as FP.
```

Comment:
We can guarantee the atomicity by delaying restoring r7 until the instruction, which requires us to update both r7 and pc at the same time. pop r7, pc is ideal for this purpose, but in order to be ready for every user of this macro, more consideration is required.